lib/ex_l7/ack.ex

```diff
@@ -0,0 +1,155 @@
+defmodule ExL7.Ack do
+  @moduledoc """
+  Provides functions to generate HL7 acknowledgement messages
+  """
+
+  import ExL7.Message.Header
+
+  @doc """
+  Returns an HL7 acknowledgement message as a string
+  """
+  def acknowledge(%ExL7.Message{} = l7_message, sequence) do
+    get_ack_hl7(l7_message, sequence, "AA")
+  end
+
+  @doc """
+  Returns an HL7 negative acknowledgement message as a string with AE as the code.
+  """
+  def error(l7_message_or_ack_config, sequence, reason \\ "")
+
+  def error(%ExL7.Message{} = l7_message, sequence, reason) do
+    get_ack_hl7(l7_message, sequence, "AE", reason)
+  end
+
+  def error(%ExL7.Ack.Config{} = ack_config, sequence, reason) do
```

Comment:
Because it's a _default_ function. Default error is purposely setting the ack_config to a default value, for default configuration. Error expects a configuration.